sklearn/neighbors/tests/test_nearest_centroid.py

```diff
@@ -3,17 +3,28 @@
 """
 import numpy as np
 import pytest
-from numpy.testing import assert_array_equal
+from numpy.testing import assert_array_almost_equal, assert_array_equal
+from scipy import sparse as sp
 
 from sklearn import datasets
 from sklearn.neighbors import NearestCentroid
+from sklearn.utils import check_random_state
 from sklearn.utils.fixes import CSR_CONTAINERS
 
 # toy sample
 X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1]]
 y = [-1, -1, -1, 1, 1, 1]
 T = [[-1, -1], [2, 2], [3, 2]]
+T_nan = np.asarray([[-1, -1], [2, 2], [3, 2]], dtype=np.float64)
```

Comment:
What is the reason to have a test for `nan`. I assume that we have some other test that was already checking this (common test) since we were not accepting those values before.

Reply:
Likely for code coverage for older commits. I removed these and code coverage is happy.